app/main.py

```diff
@@ -1 +1,34 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(
+        self,
+        name: str,
+        health: int = 100,
+        hidden: bool = False,
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+
+        self.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (
+            f"{{Name: {self.name}, Health: {self.health}, "
+            f"Hidden: {self.hidden}}}"
+        )
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, other: Animal) -> None:
+        if not isinstance(other, Herbivore) or other.hidden:
+            return
```

Comment:
let's use a direct condition: if it's carnivore - return because in the reqs it's mentioned that this method shouldn't work for Carnivore only but if we add another animal, you'll need to modify this method because for a new animal it won't work.
Also a direct condition is easier to read